specification/intelligentdataexperiences/resource-manager/Microsoft.AadCustomSecurityAttributesDiagnosticSettings/preview/2023-03-01-preview/intelligentdataexperiences.json

```diff
@@ -0,0 +1,486 @@
+  "securityDefinitions": {
+    "azure_auth": {
+      "type": "oauth2",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/providers/Microsoft.AadCustomSecurityAttributesDiagnosticSettings/operations": {
+      "get": {
+        "tags": ["Operations"],
+        "operationId": "Operations_List",
+        "x-ms-examples": {
+          "Gets the operations": { "$ref": "./examples/listOperations.json" }
+        },
+        "description": "Operation to return the list of available operations.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Successful request to get list of AadCustomSecurityAttributesDiagnosticSettings operations.",
+            "schema": { "$ref": "#/definitions/OperationsDiscoveryCollection" }
+          },
+          "default": {
+            "description": "Error, failed to get list of AadCustomSecurityAttributesDiagnosticSettings operations.",
+            "schema": { "$ref": "#/definitions/ErrorResponse" }
+          }
+        }
+      }
+    },
+    "/providers/Microsoft.AadCustomSecurityAttributesDiagnosticSettings/diagnosticSettings": {
+      "get": {
+        "tags": ["DiagnosticSettings"],
+        "operationId": "DiagnosticSettings_List",
+        "description": "Gets the active diagnostic settings list for AadCustomSecurityAttributesDiagnosticSettings.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "default": {
+            "description": "Error. Request for AadCustomSecurityAttributesDiagnosticSettings diagnostic settings has failed",
+            "schema": { "$ref": "#/definitions/ErrorResponse" }
+          },
+          "200": {
+            "description": "Successful request to get more information about AadCustomSecurityAttributesDiagnosticSettings diagnostic settings",
+            "schema": { "$ref": "#/definitions/DiagnosticSettingsResourceCollection" }
+          }
+        },
+        "x-ms-examples": {
+          "Gets the diagnostic settings": { "$ref": "./examples/listDiagnosticSettings.json" }
+        }
+      }
+    },
+    "/providers/Microsoft.AadCustomSecurityAttributesDiagnosticSettings/diagnosticSettings/{name}": {
+      "get": {
+        "tags": ["DiagnosticSettings"],
+        "operationId": "DiagnosticSettings_Get",
+        "description": "Gets the active diagnostic setting for AadCustomSecurityAttributesDiagnosticSettings.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/NameParameter"
```

Comment:
>NameParameter [](http://example.com/codeflow?start=34&length=13)

Rename this parameter to DiagnosticSettingName, as if you need to add more Resources in your RP, having just Name will conflict with all other resources also.